src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -105,6 +105,39 @@ def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=T
         self._management_resource_uri = CLOUD.endpoints.management
         self._ad_resource_uri = CLOUD.endpoints.active_directory_resource_id
 
+    def refresh_subscriptions(self):
+        from azure.cli.core.adal_authentication import AdalAuthentication
+        from azure.mgmt.resource import SubscriptionClient
+        # If no subscription is currently set CLI will throw error asking user to set a subscription or login
+        subClient = SubscriptionClient(self.get_login_credentials()[0])
+        # Figure out all tenants in the account, we will use this to look for all subscriptions
+        tenants = [x.tenant_id for x in subClient.tenants.list()]
+        # Existing logged in credential
+        tokenRetriever = subClient.config.credentials.token
+        clients = {}
+        for tenant in tenants:
+            # Attempt to pull a cred cache for each tenant
+            # if none exists auto-generates a new oauth token for this tenant
+            self.auth_ctx_factory(tenant, self._creds_cache.adal_token_cache)
+            self._creds_cache.load_adal_token_cache()
```

Comment:
I don't believe we should touch the token cache directly. 
The only time we need to touch the raw token is when we need to get refreshed tenant list by starting from the `common` tenant, after that the logic inside `_find_using_common_tenant` should be reused